submissions/vakulinina/friends-app/main.js

```diff
@@ -0,0 +1,174 @@
+};
+
+const compareByAge = (a, b) => {
+  return a.dob.age - b.dob.age
+};
+
+const sortByName = users => {
+  if (filter.nameSort === ASCENDING_ORDER) {
+    return users.sort((a, b) => compareByName(a, b))
+  } else {
+    return users.sort((a, b) => compareByName(b, a))
+  }
+};
+
+const sortByAge = users => {
+  if (filter.ageSort === ASCENDING_ORDER) {
+    return users.sort((a, b) => compareByAge(a, b))
+  } else {
+    return users.sort((a, b) => compareByAge(b, a))
+  }
+};
+
+const filterByAge = users => {
+  const [ageFrom, ageTo] = [filter.ageFrom, filter.ageTo].sort((a, b) => a - b);
+  return users.filter(({ dob: { age } }) => (age >= ageFrom && age <= ageTo))
+};
+
+const filterByGender = users => {
+  return users.filter(user => user.gender === filter.gender)
+};
+
+const doSearch = users => {
+  return users.filter(({ name: { first, last }, location: { city } }) => {
+    const fullName = `${first.toLowerCase()} ${last.toLowerCase()}`;
+    return fullName.includes(filter.searchValue) || city.toLowerCase().includes(filter.searchValue)
+  })
+};
+
+const filterUsers = () => {
+  let filteredUsers = [...allUsers];
+  if (filter.ageSort) {
+    filteredUsers = sortByAge(filteredUsers);
+  }
+  if (filter.nameSort) {
+    filteredUsers = sortByName(filteredUsers);
+  }
+  if (filter.ageFrom || filter.ageTo) {
+    filteredUsers = filterByAge(filteredUsers);
+  }
+  if (filter.gender !== 'any') {
+    filteredUsers = filterByGender(filteredUsers);
+  }
+  if (filter.searchValue) {
+    filteredUsers = doSearch(filteredUsers);
+  }
+  renderUsers(filteredUsers)
+};
+
+const setNameSortOrder = () => {
+  if (filter.nameSort === ASCENDING_ORDER) {
+    filter.nameSort = DESCENDING_ORDER;
+    sortByNameButton.classList.replace(ASCENDING_ORDER, DESCENDING_ORDER)
+  } else {
+    filter.nameSort = ASCENDING_ORDER;
+    sortByNameButton.classList.remove(DESCENDING_ORDER)
+    sortByNameButton.classList.add(ASCENDING_ORDER)
+  }
+  sortByAgeButton.classList.remove(ASCENDING_ORDER, DESCENDING_ORDER)
+  filter.ageSort = null;
+};
+
+const setAgeSortOrder = () => {
+  if (filter.ageSort === ASCENDING_ORDER) {
+    filter.ageSort = DESCENDING_ORDER;
+    sortByAgeButton.classList.replace(ASCENDING_ORDER, DESCENDING_ORDER)
+  } else {
+    filter.ageSort = ASCENDING_ORDER;
+    sortByAgeButton.classList.remove(DESCENDING_ORDER)
+    sortByAgeButton.classList.add(ASCENDING_ORDER)
+  }
+  sortByNameButton.classList.remove(ASCENDING_ORDER, DESCENDING_ORDER)
+  filter.nameSort = null;
+};
+
+const handleInput = (target) => {
+  const filters = {
+    'name-sort': setNameSortOrder,
+    'age-sort': setAgeSortOrder,
+    'age-from': () => filter.ageFrom = target.value || MIN_AGE,
+    'age-to': () => filter.ageTo = target.value || MAX_AGE,
+    'search-field': () => filter.searchValue = target.value.toLowerCase()
+  };
+
+  filters[target.id] ? filters[target.id]() : filter.gender = target.value;
```

Comment:
Don't use ternary operators like this.

Also, instead of `id` you can use [`data-*`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*). Just because you need this data for utility purposes and for searching or styling.